src/software/multithreading/observer.h

```diff
@@ -40,8 +40,10 @@ class Observer
      */
     virtual std::optional<T> popMostRecentlyReceivedValue(Duration max_wait_time) final;
 
+    virtual std::optional<T> popLeastRecentlyReceivedValue(Duration max_wait_time) final;
+
    private:
-    const size_t DEFAULT_BUFFER_SIZE = 1;
+    const size_t DEFAULT_BUFFER_SIZE = 2000;
```

Comment:
Unless I missed something, the only difference between OrderedThreadedObserver and a "regular" observer is the buffer size, and the fact that the ordered one pops the least recently added value? Since the `popLeastRecently...` function is in the `Observer` class it doesn't seem like we need an entire other class to have the "ordered" version?

Reply:
reduced code duplication